packages/strapi-plugin-i18n/admin/src/utils/mutateCTBContentTypeSchema.js

```diff
@@ -0,0 +1,59 @@
+import { has, get, omit } from 'lodash';
+import LOCALIZED_FIELDS from './localizedFields';
+
+const localizedPath = ['pluginOptions', 'i18n', 'localized'];
+
+const mutateCTBContentTypeSchema = (nextSchema, prevSchema) => {
+  // Don't perform mutations components
+  if (!has(nextSchema, localizedPath)) {
+    return nextSchema;
+  }
+
+  const isNextSchemaLocalized = get(nextSchema, localizedPath, false);
+  const isPrevSchemaLocalized = get(prevSchema, localizedPath, false);
+
+  // No need to perform modification on the schema, if the i18n feature was not changed
+  // at the ct level
+  if (isNextSchemaLocalized && isPrevSchemaLocalized) {
+    return nextSchema;
+  }
+
+  if (isNextSchemaLocalized) {
+    const attributes = Object.keys(nextSchema.attributes).reduce((acc, current) => {
+      const currentAttribute = nextSchema.attributes[current];
+
+      if (LOCALIZED_FIELDS.includes(currentAttribute.type)) {
+        const i18n = { localized: true };
+
+        const pluginOptions = currentAttribute.pluginOptions
+          ? { ...currentAttribute.pluginOptions, i18n }
+          : { i18n };
+
+        acc[current] = { ...currentAttribute, pluginOptions };
+
+        return acc;
+      }
+
+      acc[current] = currentAttribute;
+
+      return acc;
```

Comment:
Maybe we can export this algorithm in a function and add some tests? What do you think?